homeassistant/components/tibber/sensor.py

```diff
@@ -438,6 +439,61 @@ async def _fetch_data(self) -> None:
         ]["estimatedAnnualConsumption"]
 
 
+class TibberSensorElPriceExclTax(TibberSensor):
+    """Representation of a Tibber sensor for el price excluding tax."""
+
+    _attr_state_class = SensorStateClass.MEASUREMENT
+    _attr_translation_key = "electricity_price_excl_tax"
+
+    def __init__(self, tibber_home: tibber.TibberHome) -> None:
+        """Initialize the sensor."""
+        super().__init__(tibber_home=tibber_home)
+        self._last_updated: datetime.datetime | None = None
+        self._spread_load_constant = randrange(TWENTY_MINUTES)
+
+        self._attr_available = False
+        self._attr_icon = ICON
+        self._attr_unique_id = f"{self._tibber_home.home_id}-excl_tax"
+        self._model = "Price Sensor excluding tax"
+
+        self._device_name = self._home_name
+
+    async def async_update(self) -> None:
+        """Get the latest data and updates the states."""
+        now = dt_util.now()
+        if (
+            not self._tibber_home.last_data_timestamp
+            or (self._tibber_home.last_data_timestamp - now).total_seconds()
+            < 5 * 3600 + self._spread_load_constant
+            or not self.available
+        ):
+            _LOGGER.debug("Asking for new data")
+            await self._fetch_data()
+
+        elif (
+            self._tibber_home.current_price_total
+            and self._last_updated
+            and self._last_updated.hour == now.hour
+            and self._tibber_home.last_data_timestamp
+        ):
+            return
+
+        # Price energy / tax
+        # current_price_info -> {'energy': 0.0737, 'tax': 0.1719, 'total': 0.2456, 'startsAt': '2024-08-06T09:00:00.000+02:00', 'level': 'NORMAL'}
+        res = self._tibber_home.current_price_info
+        self._attr_native_value = res.get("energy")
+        self._attr_available = self._attr_native_value is not None
+        self._attr_native_unit_of_measurement = self._tibber_home.price_unit
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    async def _fetch_data(self) -> None:
+        _LOGGER.debug("Fetching data")
+        try:
+            await self._tibber_home.update_current_price_info()
```

Comment:
Why can't we use the same query as the `TibberSensorElPrice`? The price excluding tax is part of the `Price` object which is part of the `PriceInfo` object which we query in `TibberSensorElPrice` too. It should be available in the `TibberHome.info` attribute.

https://developer.tibber.com/docs/reference#price

I think we should manage the requests done so that we don't do more requests than needed to get the relevant data. Normally we use a `DataUpdateCoordinator` to solve this. Ie fetch the data with the coordinator and have both entities use the cached data.